File: javascript-agent/src/agentRunner.js
import { AgentError } from "./utils/agentError.js";
import { runOpsCoordinate } from "./agents/opsCoordinate.js";

// Intent router for this agent profile. Add new intents here for future blueprints.
export async function runAgent(task) {
  switch (task.intent) {
    case "ops.orchestrate":
      return runOpsCoordinate(task);
    default:
      throw new AgentError(
        "INTENT_UNSUPPORTED",
        `Unsupported intent: ${task.intent}`,
        false,
        400,
      );
  }
}

File: python-agent/app/agent_runner.py
from app.agents.ops_audit import run_ops_audit
from app.errors import AgentError


def run_agent(task: dict) -> dict:
    intent = task.get("intent")
    if intent == "ops.audit":
        return run_ops_audit(task)
    raise AgentError("INTENT_UNSUPPORTED", f"Unsupported intent: {intent}", False, 400)
